BUG: KoE attempting to access Underground Fireworks Shop despite its nonexistence

KolMafia version: r26272
Class: DB
Run: HC
Path: Kingdom of Exploathing
Ascensions: 32

Current Behavior:
Script is trying to buy items from the Underground Fireworks Shop, which doesn't exist in KoE. Example:

[INFO] The desert exploded so no need to build a meatcar...
Purchasing red rocket (1 @ 250)...
Unhandled redirect to place.php?whichshop=0
Desired purchase quantity not reached (wanted 1, got 0)
You need 1 more red rocket to continue.
Stack trace:
  at main (auto_pre_adv.ash:734)
[ERROR] Error running auto_pre_adv.ash, setting auto_interrupt=true
Preference auto_interrupt changed from false to true
Preference auto_interrupt changed from true to false
afterAdventureScript =>
Preference afterAdventureScript changed from scripts/autoscend/auto_post_adv.ash to
betweenBattleScript =>
Preference betweenBattleScript changed from scripts/autoscend/auto_pre_adv.ash to
breakableHandling => 1
Preference breakableHandling changed from 4 to 1
choiceAdventureScript =>
Preference choiceAdventureScript changed from scripts/autoscend/auto_choice_adv.ash to
dontStopForCounters => false
Preference dontStopForCounters changed from true to false
forbiddenStores =>
Preference forbiddenStores changed from ,3408540 to
hpAutoRecoveryTarget => 1.0
Preference hpAutoRecoveryTarget changed from -0.05 to 1.0
maximizerCombinationLimit => 0
Preference maximizerCombinationLimit changed from 100000 to 0
maximizerMRUSize => 5
Preference maximizerMRUSize changed from 0 to 5
mpAutoRecoveryTarget => 0.3
Preference mpAutoRecoveryTarget changed from -0.05 to 0.3
promptAboutCrafting => 1
Preference promptAboutCrafting changed from 0 to 1
spadingScript =>
Preference spadingScript changed from excavator.ash to
auto_interrupt detected and aborting, auto_interrupt disabled.
Stack trace:
  at auto_interruptCheck (auto_util.ash:3348)

Stack trace:
  at autoAdv (auto_adventure.ash:56)
  at LX_burnDelay (autoscend.ash:391)
[20220307.txt](https://github.com/Loathing-Associates-Scripting-Society/autoscend/files/8201698/20220307.txt)

  at process_tasks (autoscend.ash:1570)
  at doTasks (autoscend.ash:1766)
  at auto_begin (autoscend.ash:1900)
  at safe_preference_reset_wrapper (autoscend.ash:1930)
  at safe_preference_reset_wrapper (autoscend.ash:1937)
  at safe_preference_reset_wrapper (autoscend.ash:1937)
  at safe_preference_reset_wrapper (autoscend.ash:1937)

--- a/RELEASE/scripts/autoscend/iotms/mr2021.ash
+++ b/RELEASE/scripts/autoscend/iotms/mr2021.ash
@@ -454,6 +454,11 @@ boolean auto_getBattery(item target)
 
 boolean have_fireworks_shop()
 {
+	if(in_koe())
+	{
+		// can't access fireworks shop in kindom of exploathing
+		return false;
+	}
 	if(item_amount($item[Clan VIP Lounge Key]) == 0)
 	{
 		return false;

